core-util/PoolAllocator.h

```diff
@@ -45,12 +45,12 @@ class PoolAllocator {
                to satisfy the 'alignment' argument)
       * @param alignment allocation alignment in bytes (must be a power of 2, at least 4)
       */
-    PoolAllocator(void *start, size_t elements, size_t element_size, unsigned alignment = MBED_UTIL_POOL_ALLOC_DEFAULT_ALIGN);
+    PoolAllocator(void *start, size_t elements, size_t element_size, size_t alignment = MBED_UTIL_POOL_ALLOC_DEFAULT_ALIGN);
```

Comment:
Not sure why you changed this type, but in any case, there are other alignments declared as `unsigned` in this module. So you should change all of them or none of them (my advice would be to leave them as `unsigned` unless there's a really good reason to change the type).